walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,142 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    private boolean containsMandatoryAttendees = false;
+    private long MEETING_DURATION;
```

Comment:
Couple points about this. First, typically we only use all caps for constants fields. This field is not final and should use camelCase instead. Second, Using a class member variable like this is not a good practice. This is really a parameter that is being passed in for a particular call, it is not related to the class itself. You should instead pass the meeting duration around through parameters. 

Same goes for the field above. `containsMandatoryAttendees` should not be a member variable. Pass it around as a parameter instead

Reply:
I tried to clean it up a bit because I kept passing in the duration variable through nested calls that was ultimately only used once. I did it this way to avoid the parameter passing since it was only used to determine 'meetingFits()' which I overloaded as well. Same thing goes with 'containsMandatoryAttendees' because it would only be used in one function deep in the nest to determine 'isWorkable().' It also made some lines of code shorter by not having to input a duration value all the time (since it was called multiple times). Should I still revert back to passing both through all the function calls?

Reply:
What happens if two threads call `query` on the same instance of `FindMeetingQuery` at the same time?

Reply:
I see, I'm assuming the threads would overwrite eachother, maybe depending on the order in which they were called/used. I see the problem. I thought it'd make sense to make it part of the class but the issues with threading seems to cause issues